pandas/tests/io/test_rdata.py

```diff
@@ -0,0 +1,565 @@
+        read_rdata(filename, file_format="r")
+
+
+def test_read_wrong_file():
+    with pytest.raises(FileNotFoundError, match="file cannot be found"):
+        filename = os.path.join("data", "rdata", "plants_df.rda")
+        read_rdata(filename)
+
+
+def test_read_rds_non_df(datapath):
+    with pytest.raises(
+        LibrdataParserError,
+        match="Invalid file, or file has unsupported features",
+    ):
+        filename = datapath("io", "data", "rdata", "ppm_ts.rds")
+        read_rdata(filename)
+
+
+def test_read_rda_non_dfs(datapath):
+    with pytest.raises(
+        LibrdataParserError,
+        match="Invalid file, or file has unsupported features",
+    ):
+        filename = datapath("io", "data", "rdata", "env_data_non_dfs.rda")
+        read_rdata(filename)
+
+
+def test_read_not_rda_file(datapath):
+    with pytest.raises(
+        LibrdataParserError, match="The file contains an unrecognized object"
+    ):
+        filename = datapath("io", "data", "rdata", "ppm_df.csv")
+        read_rdata(filename, file_format="rda", compression=None)
+
+
+def test_bytes_read_infer_rds(datapath):
+    filename = datapath("io", "data", "rdata", "sea_ice_df.rds")
+
+    with pytest.raises(ValueError, match="Unable to infer file format from file name"):
+        with open(filename, "rb") as f:
+            read_rdata(f)
+
+
+def test_bytes_read_infer_rda(datapath):
+    filename = datapath("io", "data", "rdata", "env_data_dfs.rda")
+
+    with pytest.raises(ValueError, match="Unable to infer file format from file name"):
+        with open(filename, "rb") as f:
+            read_rdata(f)
+
+
+# URL
+
+
+@tm.network
+def test_read_rda_url():
+    url_df = DataFrame(
+        {
+            "carrier": {1: "9E", 2: "AA", 3: "AS", 4: "B6", 5: "DL"},
+            "name": {
+                1: "Endeavor Air Inc.",
+                2: "American Airlines Inc.",
+                3: "Alaska Airlines Inc.",
+                4: "JetBlue Airways",
+                5: "Delta Air Lines Inc.",
+            },
+        }
+    ).rename_axis("rownames")
+
+    url = (
+        "https://github.com/hadley/nycflights13/blob/master/data/airlines.rda?raw=true"
+    )
+    r_dfs = read_rdata(url, file_format="rda")
+
+    tm.assert_frame_equal(url_df, r_dfs["airlines"].head())
+
+
+@tm.network
+def test_read_unable_infer_format():
+    with pytest.raises(ValueError, match="Unable to infer file format from file name"):
+        url = (
+            "https://github.com/hadley/nycflights13/"
+            "blob/master/data/airlines.rda?raw=true"
+        )
+        read_rdata(url)
+
+
+@tm.network
+def test_read_wrong_url():
+    with pytest.raises(HTTPError, match="HTTP Error 404: Not Found"):
+        url = "https://example.com/data.rdata"
+        read_rdata(url)
+
+
+# S3
+
+
+@tm.network
+@td.skip_if_no("s3fs")
+def test_read_rda_s3():
+    s3 = "s3://assets.datacamp.com/production/course_1478/datasets/wine.RData"
```

Comment:
Is this a good idea?  Could trigger CI failures if datacamp makes changes.

Reply:
Good point. Can't find a usable other example unless we push to pandas' S3 account.